galaxy/Dockerfile

```diff
@@ -110,77 +48,106 @@ RUN curl -s -L https://github.com/conda-forge/miniforge/releases/download/$MINIF
+    # Install galaxy client
+    && cd $GALAXY_ROOT_DIR && ./scripts/common_startup.sh \
+    && . $GALAXY_VIRTUAL_ENV/bin/activate \
+    && pip install "weasyprint>=61.2" watchdog --index-url https://wheels.galaxyproject.org/simple --extra-index-url https://pypi.python.org/simple \
+    && deactivate \
+    && cd config && find . -name 'node_modules' -type d -prune -exec rm -rf '{}' + \
+    && find . -name '.cache' -type d -prune -exec rm -rf '{}' + \
+    && chown -R $GALAXY_USER:$GALAXY_USER $GALAXY_ROOT_DIR $GALAXY_VIRTUAL_ENV \
+    && rm -rf $GALAXY_ROOT_DIR/client/node_modules/ $GALAXY_VIRTUAL_ENV/src/ \
+    && /usr/bin/common_cleanup.sh
 
 
-ADD ./reports.yml.sample $GALAXY_CONFIG_DIR/reports.yml
-ADD sample_tool_list.yaml $GALAXY_HOME/ephemeris/sample_tool_list.yaml
+FROM ubuntu:22.04 AS galaxy_cluster_base
 
-# Activate Interactive Tools during runtime
-ADD ./tools_conf_interactive.xml.sample $GALAXY_INTERACTIVE_TOOLS_CONFIG_FILE
+ENV GALAXY_ROOT_DIR=/galaxy-central \
+    GALAXY_VIRTUAL_ENV=/galaxy_venv \
+    GALAXY_LOGS_DIR=/home/galaxy/logs \
+    GALAXY_CONFIG_DIR=/etc/galaxy \
+    GALAXY_USER=galaxy \
+    GALAXY_UID=1450 \
+    GALAXY_GID=1450 \
+    GALAXY_HOME=/home/galaxy \
+    GALAXY_CONDA_PREFIX=/tool_deps/_conda \
+    EXPORT_DIR=/export \
+    DEBIAN_FRONTEND=noninteractive \
+    PG_VERSION=15
 
-RUN curl -o $GALAXY_CONFIG_TOOL_DATA_TABLE_CONFIG_PATH \
-    -L https://raw.githubusercontent.com/galaxyproject/usegalaxy-playbook/8adb1f82c94fe95b09df2a2816440ce2420b7d39/env/main/files/galaxy/config/tool_data_table_conf.xml \
-    && chown $GALAXY_USER:$GALAXY_USER $GALAXY_CONFIG_DIR/reports.yml $GALAXY_HOME/ephemeris/sample_tool_list.yaml $GALAXY_INTERACTIVE_TOOLS_CONFIG_FILE $GALAXY_CONFIG_TOOL_DATA_TABLE_CONFIG_PATH \
-    && chmod 0644 $GALAXY_CONFIG_DIR/reports.yml $GALAXY_HOME/ephemeris/sample_tool_list.yaml $GALAXY_INTERACTIVE_TOOLS_CONFIG_FILE $GALAXY_CONFIG_TOOL_DATA_TABLE_CONFIG_PATH
+ENV GALAXY_CONFIG_FILE=$GALAXY_CONFIG_DIR/galaxy.yml \
+    GALAXY_CONFIG_JOB_CONFIG_FILE=$GALAXY_CONFIG_DIR/job_conf.xml \
+    GALAXY_CONFIG_JOB_METRICS_CONFIG_FILE=$GALAXY_CONFIG_DIR/job_metrics_conf.yml \
+    GALAXY_CONFIG_TUS_UPLOAD_STORE=$EXPORT_DIR/tus_upload_store \
+    GALAXY_CONFIG_INTERACTIVETOOLS_MAP=$EXPORT_DIR/galaxy-central/database/interactivetools_map.sqlite \
+    GRAVITY_CONFIG_FILE=$GALAXY_CONFIG_DIR/gravity.yml \
+    GALAXY_POSTGRES_UID=1550 \
+    GALAXY_POSTGRES_GID=1550 \
+    # Define the default postgresql database path
+    PG_DATA_DIR_DEFAULT=/var/lib/postgresql/$PG_VERSION/main/ \
+    PG_CONF_DIR_DEFAULT=/etc/postgresql/$PG_VERSION/main/ \
+    PG_DATA_DIR_HOST=$EXPORT_DIR/postgresql/$PG_VERSION/main/
+
+ADD ./common_cleanup.sh /usr/bin/common_cleanup.sh
+RUN chmod +x /usr/bin/common_cleanup.sh
+
+RUN echo "force-unsafe-io" > /etc/dpkg/dpkg.cfg.d/02apt-speedup \
```

Comment:
If it's not the final container we do not need to cleanup isn't it? That is one of the advantages of a multi stage build. It makes the file more readable.

Reply:
Yes true. I will do the changes.